src/app/modules/group/components/associated-activity/associated-activity.component.ts

```diff
@@ -67,19 +79,34 @@ export class AssociatedActivityComponent implements OnDestroy, ControlValueAcces
     this.activityChanges.complete();
   }
 
-  writeValue(rootActivityId: ActivityId|null): void {
-    this.activityChanges.next({ id: rootActivityId, triggerChange: false });
+  writeValue(rootActivity: NoActivity|NewActivity|ExistingActivity): void {
+    this.activityChanges.next({ activity: rootActivity, triggerChange: false });
   }
 
-  registerOnChange(fn: (value: ActivityId|null) => void): void {
+  registerOnChange(fn: (value: NoActivity|NewActivity|ExistingActivity) => void): void {
     this.onChange = fn;
   }
 
   registerOnTouched(_fn: () => void): void {
   }
 
   onRemove(): void {
-    if (this.rootActivity === null || this.state !== 'ready') throw new Error('Unexpected: tried to remove root activity when not ready');
-    this.activityChanges.next({ id: null, triggerChange: true });
+    if (this.rootActivity.type === 'no-activity' || this.state !== 'ready') {
+      throw new Error('Unexpected: tried to remove root activity when not ready or no prior activity');
+    }
+
+    this.activityChanges.next({ activity: { type: 'no-activity' }, triggerChange: true });
+  }
+
+  setRootActivity(activty: AddedContent<ItemType>): void {
+    if (this.rootActivity.type !== 'no-activity' || this.state !== 'ready') {
+      throw new Error('Unexpected: tried to set a root activty when not ready or already set activity');
+    }
+
+    if (activty.id !== undefined) {
+      this.activityChanges.next({ activity: { type: 'existing-activity', id: activty.id }, triggerChange: true });
```

Comment:
Don't you know the name and type from the search service? If so, it is probably not useful to refetch them.

Reply:
fixed